### PR TITLE
Vagrantfile sync-up

### PR DESCRIPTION
**Pull request by arm4b**
Use `ubuntu/trusty64` box by default, lift up default box memory and changes about Vagrant plugins requirements.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-  # All Vagrant configuration is done here. The most common configuration
-  # options are documented and commented below. For a complete reference,
-  # please see the online documentation at vagrantup.com.
-
-  config.vm.hostname = 'stackstorm-berkshelf'
+# Autoinstall for vagrant-omnibus plugin.
+# See https://github.com/chef/vagrant-omnibus
+unless Vagrant.has_plugin?('vagrant-omnibus')
+  system('vagrant plugin install vagrant-omnibus') || exit!
```
Comment (shortdudey123) [inline Vagrantfile]: hmm, is auto installing the best thing here?
IMO its better to tell the user to install it instead of forcibly installing it for them

Comment (arm4b) [reply Vagrantfile]: Yeah, defaults to get things running fast.
My thinking that it's not ideal, but it's not possible to run this particular `Vagrantfile` without installing them. So it's balance.

If you have strong opinion that we shouldn't do that and it's kind of super crazy, - I'm OK to change that ;)

Comment (shortdudey123) [reply Vagrantfile]: I would say its better to give the user the command to do it in a log output then do it for them.  Some users may not want the plugins installed in their environment.

Comment (arm4b) [reply Vagrantfile]: I found we have another abandoned Vagrantfile PR.
So I took some good things from that place https://github.com/StackStorm/chef-stackstorm/pull/21/files#diff-23b6f443c01ea2efcb4f36eedfea9089R4, thanks to @jvrplmlmn

**Comment by shortdudey123** [comment]
Looking better :)
Can you fix the rubocop offense and maybe squash the commits since you have a merge master commit?

**Comment by arm4b** [comment]
Thanks @shortdudey123!

**Comment by shortdudey123** [comment]
@armab the squash was missing when you merged :(

**Comment by arm4b** [comment]
@shortdudey123 It's there!
See: https://github.com/StackStorm/chef-stackstorm/commits/master

That GitHub magic button :)

**Comment by shortdudey123** [comment]
oh gotcha, i don't use that feature at all haha

**Pull request merged**